addons/mail/static/src/models/emoji_registry.js

```diff
@@ -2,109 +2,52 @@
 
 import { registerModel } from '@mail/model/model_core';
 import { many } from '@mail/model/model_field';
-import { insertAndReplace } from '@mail/model/model_field_command';
+import { insert } from '@mail/model/model_field_command';
+import { emojiCategoriesData, emojiData } from '@mail/models_data/emoji_data';
 
 registerModel({
     name: 'EmojiRegistry',
     identifyingFields: ['messaging'],
+    lifecycleHooks: {
+        _created() {
+            this._populateFromEmojiData();
+        },
+    },
+    recordMethods: {
+        _populateFromEmojiData() {
+            this.models['EmojiCategory'].insert(emojiCategoriesData.map(category => {
+                return {
+                    categoryName: category.name,
+                    categoryTitle: category.title,
+                    categorySortId: category.sortId,
+                };
+            }));
+            this.models['Emoji'].insert(emojiData.map(emoji => {
+                return {
+                    unicode: emoji.codepoints,
+                    sources: [...emoji.shortcodes, ...emoji.emoticons],
+                    description: emoji.name,
+                    emojiCategories: insert([
+                        { categoryName: emoji.category },
+                    ]),
```

Comment:
the name of some fields in Emoji should be adapted to match the same naming choice as `emoji_data`, like `Emoji/description` becoming `Emoji/name`.